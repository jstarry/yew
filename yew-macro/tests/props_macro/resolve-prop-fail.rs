use yew::prelude::*;

#[derive(PartialEq, Properties)]
struct Props {}

struct MyComp;
impl Component for MyComp {
    type Message = ();
    type Properties = Props;

    fn create(_: &Context<Self>) -> Self {
        unimplemented!()
    }

    fn view(&self, _: &Context<Self>) -> Html {
        unimplemented!()
    }
}

trait NotAComponent {
    type Properties;
}

struct MyNotAComponent;
impl NotAComponent for MyNotAComponent {
    type Properties = ();
}

fn compile_fail() {
    yew::props!(Vec<_> {});
    yew::props!(MyComp {});
    yew::props!(MyNotAComponent::Properties {});
}

fn main() {}
