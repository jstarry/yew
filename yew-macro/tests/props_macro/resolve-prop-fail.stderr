error[E0277]: the trait bound `std::vec::Vec<_>: yew::html::Properties` is not satisfied
  --> $DIR/resolve-prop-fail.rs:30:17
   |
30 |     yew::props!(Vec<_> {});
   |                 ^^^ the trait `yew::html::Properties` is not implemented for `std::vec::Vec<_>`

error[E0277]: the trait bound `MyComp: yew::html::Properties` is not satisfied
  --> $DIR/resolve-prop-fail.rs:31:17
   |
31 |     yew::props!(MyComp {});
   |                 ^^^^^^ the trait `yew::html::Properties` is not implemented for `MyComp`

error[E0277]: the trait bound `MyNotAComponent: yew::component::Component` is not satisfied
  --> $DIR/resolve-prop-fail.rs:32:17
   |
32 |     yew::props!(MyNotAComponent::Properties {});
   |                 ^^^^^^^^^^^^^^^ the trait `yew::component::Component` is not implemented for `MyNotAComponent`
